scrape/manager.go

```diff
@@ -100,6 +100,40 @@ func (mc *MetadataMetricsCollector) Collect(ch chan<- prometheus.Metric) {
 	}
 }
 
+// ManagerGetter is an interface to represent a ReadyManager.
+type ManagerGetter interface {
```

Comment:
We can't use `Manager` because is already part of this package, and `Managable`.. Well, you already pointed that one out. I'm happy to listen to other suggestions as this is the best that occurred to me.

Reply:
Right so the interface would be `Manageable` and it would have a function that returns a `*Manager`. Not really worth spending time on right now though.

Reply:
> vote for ReadyManager and rename struct to ReadyScrapeManager
Love this! Thanks for the suggestion.

> Also ideally we can remove this interface and use struct directly. Reason: We have just one, simple implementation.

I would like to keep the interface around. The interface only exposes the `Get` method which signals that those are the needs other packages need to support.

> Comment would be nice anyway, what it does
Sorry! I'm not sure I follow this comment, are you implying I should also add the same comment to the interface? The `Get` method already describes the behaviour in your second comment, adding it to the interface felt repetitive. Happy to do so if you still disagree!

Reply:
If you feel it's verbose now with new naming then all goo.